In [9]:
import numpy as np

$$C_n^k = \frac{n!}{k!(n - k)!}$$

In [12]:
from math import factorial

In [13]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [14]:
combinations(100, 85)

253338471349988640

$$p(k=85) = C_n^kp^kq^{n-k} = C_{100}^{85}p^{85}q^{15} = \frac{100!}{85!(100-85)!} \cdot 0.8^{85} \cdot 0.2^{15} = 253338471349988640 \cdot 5.7896 \cdot 3.2768 = 0.0480$$

$$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$

**где** $λ = np$.

In [ ]:
n = 5000
m = 2
p = 0.0004
lambda_ = n * p

(lambda_**m / np.math.factorial(m)) * np.exp(-lambda_)

Применим формулу Пуассона, зная, что $m = 0$ и $λ = np = 5000 \cdot 0.0004 = 2$:

$$P_{0} \approx \frac{\lambda^m}{m!}e^{-\lambda} \approx \frac{2^{0}}{0!}e^{-2} \approx  0.1353$$

Применим формулу Пуассона, зная, что $m = 2$ и $λ = np = 5000 \cdot 0.0004 = 2$:

$$P_{2} \approx \frac{\lambda^m}{m!}e^{-\lambda} \approx \frac{2^{2}}{2!}e^{-2} \approx  0.2707$$

$$p(k=70) = C_n^kp^kq^{n-k} = C_{144}^{70}p^{70}q^{74} = \frac{144!}{70!(144-70)!} \cdot 0.5^{70} \cdot 0.5^{74} = 1400749509083708812152608657850239525991600 \cdot 8.4703 \cdot 5.2939 = 0.0628$$

$$ P = (\frac{7}{10} \cdot \frac{6}{9}) \cdot (\frac{9}{11}\cdot \frac{8}{10}) \approx 0.3055 $$

$$ P(A) = 1- ((\frac{3}{10} \cdot \frac{2}{9}) \cdot (\frac{2}{11}\cdot \frac{1}{10})) \approx 0.9988 $$